lib/ae_mdw/aex9.ex

```diff
@@ -39,6 +39,16 @@ defmodule AeMdw.Aex9 do
 
   @type amounts :: map()
 
+  @spec invalid_number_of_holders() :: String.t()
+  def invalid_number_of_holders() do
+    "Invalid number of holders"
+  end
+
+  @spec invalid_holder_balance() :: String.t()
+  def invalid_holder_balance() do
+    "Invalid holder balance"
+  end
```

Comment:
What's the purpose of these functions? Can we use module a module attribute or directly include it in the code instead?

Reply:
They are used in different modules, so a module attribute won't do. As for directly in the code - there could be a hard to notice difference, and a change in the future might be incomplete. Having them in a single place has a better guarantee for consistency. 

The other way we could do it is with macros, that just returns the strings - that way we could also use them in pattern matches if we want, the downside is that we have to `require` the module that contains them (I personally don't mind this, rather I prefer it).

Reply:
Got it, it's being reused in several places, so perhaps we can do something like `def invalid_reason(:holder_balance)` and have dialyzer check if the input is either `:holder_balance | :other_reason`

Reply:
Do we gain something doing it that way though? Because now instead of a compile-time error, we potentially have a static analysis one?

Reply:
Functions should describe what these values represent, not what these values are. Which is why we write
```elixir
def maximum_transactions, do: 77

if length(transactions) >= maximum_transactions() do ...
```
and not
```elixir
def seventy_seven, do: 77

if length(transactions) >= seventy_seven() do ...
```

These are reasons for a contract being invalid, not an invalid holder's balance or an invalid number of holders

Reply:
Going with that, the function can be named `invalid_holder_balance_message`, that way it would describe what the value represents. Even if that was not possible though, I think sacrificing the intended use of a language features to gain better guarantees (compile-time checks are always better than dialyzer ones) is worth it.

In any case, I see you feel strongly about it and I don't think in this particular case it would provide much additional value, I will switch it as suggested.

Reply:
I agree compile-time errors are way superior to external static checks like dialyzer, so I wouldn't mind either way `invalid_holder_balance_reason` is fine to me